assignment4/assignment4.cpp

```diff
@@ -2,26 +2,40 @@
 #include <math.h>
 #include <vector>
 
-class IslandCounter{
-	void collapse_island(std::vector< std::vector<bool> > &map, int i, int j){
-		if(!map[i][j]) return;
+class IslandCounter {
+	std::vector< std::vector<bool> >  map;
+	void collapse_island ( int i, int j) {
+		if (!map[i][j]) return;
 
 		map[i][j] = false;
-		collapse_island( map, std::min(i+1, (int) (map.size()-1)), j);
-		collapse_island( map, std::max(i-1, 0), j);
-		collapse_island( map, i, std::min(j+1,(int) (map[i].size()-1)));
-		collapse_island( map, i, std::max(j-1,0));
-		
+		int end_row = std::min(i+1, (int) (map.size()-1));
```

Comment:
If you start use some particular code style all your code in the one source file should be formatting according to the one code style.
You've started adding spaces around the operations, but here are some missing spaces.